yaksh/views.py

```diff
@@ -3191,3 +3191,70 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def course_forum(request, course_id):
+    user = request.user
+    course = get_object_or_404(Course, id=course_id)
```

Comment:
Add a condition to check if the user is either course creator, teacher or student.